tests/deliver_to_wrong_chain.go

```diff
@@ -0,0 +1,92 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+func DeliverToWrongChainGinkgo() {
+	DeliverToWrongChain(&network.LocalNetwork{})
+}
+
+func DeliverToWrongChain(network network.Network) {
+	var (
+		teleporterMessageID *big.Int
+	)
+
+	subnets := network.GetSubnetsInfo()
+	subnetAInfo := subnets[0]
+	subnetBInfo := subnets[1]
+	teleporterContractAddress := network.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	subnetATeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetAInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+	subnetBTeleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(
+		teleporterContractAddress, subnetBInfo.ChainRPCClient,
+	)
+	Expect(err).Should(BeNil())
+
+	//
+	// Get the expected teleporter message ID for Subnet B
+	//
+	teleporterMessageID, err = subnetATeleporterMessenger.GetNextMessageID(&bind.CallOpts{}, subnetBInfo.BlockchainID)
+	Expect(err).Should(BeNil())
+
+	//
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to a chain other than Subnet B
+	//
+	ctx := context.Background()
+
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: ids.Empty, // Some other chain ID
+		DestinationAddress: fundedAddress,
```

Comment:
nit: can we note this is a random address, or just use address(0)

Reply:
Added a dummy address